models/DataObject/ClassDefinition/LinkGeneratorInterface.php

```diff
@@ -15,15 +15,13 @@
 
 namespace Pimcore\Model\DataObject\ClassDefinition;
 
-use Pimcore\Model\DataObject\Concrete;
-
 interface LinkGeneratorInterface
 {
     /**
-     * @param Concrete $object
+     * @param object $object
      * @param array $params
      *
      * @return string
      */
-    public function generate(Concrete $object, array $params = []): string;
+    public function generate(object $object, array $params = []): string;
```

Comment:
@ckogler-elements BC break? As existing implementations will break with Incompatible Declaration error.

Reply:
Yes, but i don't see a way around it...

Reply:
Is it really a BC break? Concrete is a specific object. So it should still work, right?

Reply:
Like i said, it will throw declaration error as `generate()` implementations needs to change the param type hint as well. https://php.watch/versions/8.0/lsp-errors

Reply:
It is, because parameter types are contravariant. So the implementation is allowed to be *less specific*. But if you change the interface to be *less specific*, all existing implementations are *more specific* (except they left out the parameter type), which is not allowed and thus a BC break.

Reply:
Yes you are right. I test it. I think with class and subclasses it should work. But object and class doesn't.
See: https://www.php.net/manual/en/language.oop5.variance.php

Reply:
@ckogler-elements This means you can revert this change. You can create LinkGenerator classes with object type hints anyway. This works.
You should add only a comment to change this interface in Pimcore 11.

Reply:
@blankse : thx - that works :-)
@dvesh3 : Changed it back and added a hint to the documentation